doc/man3/OSSL_PROVIDER.pod

```diff
@@ -0,0 +1,110 @@
+=pod
+
+=head1 NAME
+
+OSSL_PROVIDER, OSSL_load_provider, OSSL_unload_provider,
+OSSL_get_provider_param_types, OSSL_get_provider_params,
+OSSL_add_provider - provider routines
+
+=head1 SYNOPSIS
+
+ #include <openssl/provider.h>
+
+ typedef struct ossl_provider_st OSSL_PROVIDER;
+
+ OSSL_PROVIDER *OSSL_load_provider(OPENSSL_CTX *, const char *name);
+ int OSSL_unload_provider(OSSL_PROVIDER *prov);
+
+ const OSSL_ITEM *OSSL_get_provider_param_types(OSSL_PROVIDER *prov);
+ int OSSL_get_provider_params(OSSL_PROVIDER *prov, const OSSL_PARAM params[]);
+
+ int OSSL_add_provider(OPENSSL_CTX *,
+                       const char *name, ossl_provider_init_fn *init_fn);
+
+=head1 DESCRIPTION
+
+C<OSSL_PROVIDER> is a type that holds internal information about
+implementation providers (see L<provider(7)> for information on what a
+provider is).
+A provider can be built in to the application or the OpenSSL
+libraries, or can be a loadable module.
+The functions described here handle both forms.
+
+=head2 Functions
+
+OSSL_add_provider() is used to add a built in provider in the given
```

Comment:
add a built in provider **to** the given library context

Reply:
Yup